### PR TITLE
Adding support for CS2

### PR DESCRIPTION
**Pull request by cokenyo**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,25 @@
 module github.com/csconfederation/demoScrape2
 
-go 1.16
+go 1.21
+
+toolchain go1.21.3
 
 require (
-	github.com/markus-wa/demoinfocs-golang/v2 v2.10.0
+	github.com/markus-wa/demoinfocs-golang/v4 v4.0.0-beta.5.0.20231026145227-3c4b79fde38c
```
Comment (martig3) [inline go.mod]: looks like this can be bumped to `v4.0.0`

`github.com/markus-wa/demoinfocs-golang/v4 v4.0.0`

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+	github.com/markus-wa/gobitread v0.2.3 // indirect
+	github.com/markus-wa/godispatch v1.4.1 // indirect
+	github.com/markus-wa/ice-cipher-go v0.0.0-20230901094113-348096939ba7 // indirect
+	github.com/markus-wa/quickhull-go/v2 v2.2.0 // indirect
+	github.com/oklog/ulid/v2 v2.1.0 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	golang.org/x/sys v0.0.0-20220715151400-c0bba94af5f8 // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+	google.golang.org/protobuf v1.31.0 // indirect
+)
```
Comment (martig3) [inline go.mod]: this can be removed after updating dependencies I believe

**Review by martig3 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (cokenyo) [reply go.mod]: Updated

File: go.mod (hunk 2 above)
Comment (cokenyo) [reply go.mod]: Re-tidying updated some versions and took away xerrors after swapping to the non-beta version. But these were still generated. Should this avoid going to the production version?

**Pull request merged**